ibis/expr/operations/generic.py

```diff
@@ -275,11 +275,24 @@ class SimpleCase(Value):
     results: VarTuple[Value]
     default: Value
 
-    shape = rlz.shape_like("base")
-
-    def __init__(self, cases, results, **kwargs):
+    def __init__(self, base, cases, results, default):
         assert len(cases) == len(results)
-        super().__init__(cases=cases, results=results, **kwargs)
+
+        for case in cases:
+            if not rlz.comparable(base, case):
+                raise TypeError(
```

Comment:
Can you leave this as is? I don't see why these changes are necessary.

Reply:
consider `ibis.literal("foo").cases((1, "small"), (4, "big"))`. Currently, this only errors at runtime. This check makes this error at construction time.

I can put this in a separate PR, but I think this is a good improvement I'd like to see, what do you think?

Reply:
Any changes that affect validation and/or type checking or construction of operations need to go in separately from API changes. If you want to make this change please put it in a separate PR.

Reply:
PR ready for review at https://github.com/ibis-project/ibis/pull/9559, thanks!